libnvme-rs/src/nvme_namespaces.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{error, parse_value};
+use error::NvmeError;
+use glob::glob;
+use std::{os::unix::fs::FileTypeExt, path::Path};
+
+/// NvmeDevices are devices that are already connected to the kernel
+/// they have not interaction with the fabric itself. Notice that a
+/// nvme device, for the post part is a subsystem + nsid.
+
+#[derive(Debug, Default)]
+pub struct NvmeDevice {
+    /// device path of the device
+    pub path: String,
+    /// the device model defined by the manufacturer
+    model: String,
+    /// serial number of the device
+    serial: String,
+    /// the size in bytes
+    size: u64,
+    /// the UUID for the device
+    uuid: String,
+    /// the world wide name of the device typically wwn.uuid
+    wwid: String,
+    /// the namespace id
+    nsid: u64,
+    /// firmware revision
+    fw_rev: String,
+    /// the nqn of the subsystem this device instance is connected to
+    pub subsysnqn: String,
+}
+
+impl NvmeDevice {
+    /// Construct a new NVMe device from a given path. The [struct.NvmeDevice]
+    /// will fill in all the details defined within the structure or return an
+    /// error if the value for the structure could not be found.
+    fn new(p: &Path) -> Result<Self, NvmeError> {
+        let name = p.file_name().unwrap().to_str().unwrap();
+        let devpath = format!("/sys/block/{}", name);
+        let subsyspath = format!("/sys/block/{}/device", name);
```

Comment:
This actually gives the path of the controller right?

Reply:
Yes.